ui/src/main/webapp/wizards/ldap/stages/network-settings.js

```diff
@@ -32,87 +43,163 @@ const testConnect = (conn) => ({
+
+    const { messages, ...errors } = groupErrors([
+      'hostname',
+      'port',
+      'encryption'
+    ], this.props.errors)
+
+    const isPortInvalid = configs.port === undefined || configs.port < 0 || configs.port > 65535
+
+    const missingHost = configs.connectionInfo === undefined || Object.keys(configs.connectionInfo).length === 0
+
+    return (
+      <div>
+        <Mount
+          on={setDefaults}
+          port={636}
+          encryption='ldaps'
+          loadbalancing='roundRobin' />
+
+        <Title>LDAP Network Settings</Title>
+        <Description>
+          To establish a connection to the remote LDAP store, we need the hostname of the
+          LDAP machine, the port number that the LDAP service is running on, and the
+          encryption method. Typically, port 636 uses LDAPS encryption and port 389 uses
+          StartTLS.  Adding more than one host creates a cluster.  The cluster connections can
+          be load balanced between the hosts (round-robin) or can be treated as a failover cluster
+          where the first host is considered the primary host.  The configuration of each host
+          in the cluster is assumed to be the same (i.e. bind user, directory settings, etc).
+        </Description>
+
+        <Body>
+          <Select
+            value={configs.encryption}
+            errorText={errors.encryption}
+            onEdit={onEdit('encryption')}
+            label='Encryption Method'
+            options={[ 'none', 'ldaps', 'startTls' ]} />
+
+          <Hostname
+            value={configs.hostname}
+            errorText={errors.hostname}
+            onEdit={onEdit('hostname')}
+            autoFocus />
+
+          <Port
+            value={configs.port}
+            errorText={isPortInvalid ? getFriendlyMessage('INVALID_PORT_RANGE') : errors.port}
+            onEdit={onEdit('port')}
+            options={[389, 636]} />
+
+          <Select
+            value={configs.loadbalancing}
+            onEdit={onEdit('loadbalancing')}
+            label='Load Balancing Algorithm'
+            options={[ 'roundRobin', 'failover' ]} />
+
+          {messages.map((msg, i) => <Message key={i} {...msg} />)}
+
+          <RaisedButton
+            primary
+            style={{margin: '0 auto', marginBottom: '30px', marginTop: '10px', display: 'block'}}
+            label='Add Host'
+            disabled={isPortInvalid}
             onClick={() => {
               onStartSubmit()
               client.query(testConnect({
                 hostname: configs.hostname,
                 port: configs.port,
                 encryption: configs.encryption
-              }))
-                .then(() => {
-                  onEndSubmit()
-                  next('bind-settings')
-                })
-                .catch((err) => {
-                  onEndSubmit()
-                  onError(err)
+              })).then((result) => {
+                connectionInfo.push([configs.hostname, configs.port])
```

Comment:
`.push()` will mutate the original list, you might want `.concat()` instead. Also can we append an object instead? `{hostname: configs.hostname, port: configs.port}`